autogen/agentchat/multimodal_conversable_agent.py

```diff
@@ -0,0 +1,118 @@
+from typing import Callable, Dict, List, Optional, Tuple, Union
+
+from autogen import oai
+from autogen.img_utils import gpt4v_formatter
+
+from .agent import Agent
+from .conversable_agent import ConversableAgent
+
+try:
+    from termcolor import colored
+except ImportError:
+
+    def colored(x, *args, **kwargs):
+        return x
+
+
+DEFAULT_LMM_SYS_MSG = """You are a helpful AI assistant.
+You can also view images, where the "<image i>" represent the i-th image you received."""
+
+
+class MultimodalConversableAgent(ConversableAgent):
+    def __init__(
+        self,
+        name: str,
+        system_message: Optional[Tuple[str, List]] = DEFAULT_LMM_SYS_MSG,
+        is_termination_msg=None,
+        *args,
+        **kwargs,
+    ):
+        """
+        Args:
+            name (str): agent name.
+            system_message (str): system message for the ChatCompletion inference.
+                Please override this attribute if you want to reprogram the agent.
+            **kwargs (dict): Please refer to other kwargs in
+                [ConversableAgent](conversable_agent#__init__).
+        """
+        super().__init__(
+            name,
+            system_message,
+            is_termination_msg=is_termination_msg,
+            *args,
+            **kwargs,
+        )
+
+        self.update_system_message(system_message)
+        self._is_termination_msg = (
+            is_termination_msg if is_termination_msg is not None else (lambda x: x.get("content")[-1] == "TERMINATE")
```

Comment:
if we use the new sys msg, should we incorporate origianl sys msg for gpt? I think we may at least add the last sentence "Reply "TERMINATE" in the end when everything is done." to the new sys msg to make this termination condition work?

Reply:
The multimodal conversable agent inherits from the "ConversableAgent" that has the default system message "You are a helpful AI Assistant."   Hence, I think the "TERMINATE" clause is not necessary here (correct me if I am wrong). If the user wants to create a Multimodal Assistant Agent, then it would requires the long system message from the Assistant Agent.

Reply:
I see, you are implying that user may use multimodal conversable agent as user proxy?

Reply:
I'm thinking if we only expect multimodal assistant agent, then we can directly do it instead of multimodal conversable agent? Not sure which one is better, but I'm leaning towards the multimodal assistant agent unless there are use cases where we need to do multimodal user proxy agent